c7n/resources/kms.py

```diff
@@ -64,6 +64,30 @@ class Meta(object):
 
     resource_type = Meta
 
+@Key.filter_registry.register('key-rotation-status')
+class KeyRotationStatus(ValueFilter):
+
+    schema = type_schema('key-rotation-status', rinherit=ValueFilter.schema)
+
+    def process(self, resources, event=None):
+
+        def _key_rotation_status(resource):
+            client = local_session(self.manager.session_factory).client('kms')
+            resource['KeyRotationEnabled'] = client.get_key_rotation_status(
+                KeyId=resource['KeyId'])
+
+        with self.executor_factory(max_workers=2) as w:
+            query_resources = [
+                r for r in resources if 'KeyRotationEnabled' not in r]
+            self.log.debug("Querying %d kms-keys' rotation status" % len(query_resources))
+            list(w.map(_key_rotation_status, query_resources))
+
+        matched = []
+        for r in resources:
+            if self.match(r['KeyRotationEnabled']):
```

Comment:
fwiw you could one liner these 5 with
return [r for r in resources if self.matched(r['KeyRotationEnabled')